src/zip_code_tree.hpp

```diff
@@ -691,4 +872,758 @@ struct iterator_traits<vg::ZipCodeTree::reverse_iterator>{
 
 }
 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/// Implementations for the templated functions using Minimizers since the definition is in the minimizer_mapper
+//TODO: This really shouldn't be in the hpp file
+
+namespace vg {
+    using namespace std;
+
+template<typename Minimizer>
+void ZipCodeForest::fill_in_forest(const vector<Seed>& seeds, const VectorView<Minimizer>& minimizers,
+                                   const SnarlDistanceIndex& distance_index, size_t gap_distance_limit,
+                                   size_t distance_limit) {
+#ifdef DEBUG_ZIP_CODE_TREE
+    cerr << "Make a new forest with " << seeds.size() << " seeds with distance limit " << distance_limit << endl;
+    for (auto& x : seeds) {
+        cerr << x.pos << endl;
+    }
+    cerr << endl;
+#endif
+    if (seeds.size() == 0) {
+        return;
+    }
+
+    /*
+    Make a ZipCodeForest
+    Takes a vector of seeds and their minimizers and fills in the forest
+
+    The zip forest is made by sorting the seeds along chains/snarls, then adding each seed, 
+    snarl/chain boundary, and distance to zip_code_tree.
+
+    Sorting and tree-making is done at the same time, in a depth-first traversal of the snarl tree.
+    Sorting is done per node in the snarl tree; seeds along a chain are sorted to be ordered along
+    a chain, and seeds in a snarl are sorted by the child of the snarl that they are on.
+
+    "Intervals" of seeds in the sort order are used to keep track of the location on the snarl tree.
+    An interval represents a range of seeds that are all on the same snarl tree structure.
+    After sorting an interval at one depth, sub-intervals representing the children can be found.
+    
+    Intervals are stored in a stack. The algorithm starts with an interval for each child of the
+    root snarl. An interval is popped from the stack. Any incomplete snarls or chains that the
+    interval is not a child of must be completed. Then, the snarl or chain that the interval
+    represents is added to the zip tree, along with any relevant distances. Intervals representing 
+    the children of the snarl or chain are found and added to the stack. This repeats until the
+    stack is empty.
+
+    */
```

Comment:
This is an important high-level overview of the whole zip code tree build process, but it's buried in this function. Maybe it wants to be in the header somewhere, before the definitions of all the functions that represent pieces of this algorithm?